fission/src/ui/panels/configuring/MatchModeConfigPanel.tsx

```diff
@@ -0,0 +1,286 @@
+import { Box } from "@mui/material"
+import { usePanelControlContext } from "@/ui/helpers/UsePanelManager"
+import MatchMode from "@/systems/MatchMode"
+import { Global_AddToast } from "@/ui/components/GlobalUIControls"
+import { useModalControlContext } from "@/ui/helpers/UseModalManager"
+import Button from "@/ui/components/Button"
+
+export interface MatchModeConfig {
+    id: string // Required
+    name: string // Required
+    autonomousTime: number // Optional, defaults to 15
+    teleopTime: number // Optional, defaults to 135
+    endgameTime: number // Optional, defaults to 20
+}
+
+function MatchConfigSelected(config: MatchModeConfig, openModal: (modalName: string) => void) {
+    if (MatchMode.getInstance().isMatchEnabled()) {
+        Global_AddToast?.(
+            "error",
+            "Match Mode Already Running",
+            "You can't modify the match mode ruleset while a match is running"
+        )
+        return
+    }
+
+    MatchMode.getInstance().setMatchModeConfig(config)
+
+    MatchMode.getInstance().start(openModal)
+}
+
+interface ItemCardProps {
+    id: string
+    name: string
+    primaryOnClick: () => void
+    secondaryOnClick?: () => void
+}
+
+const ItemCard: React.FC<ItemCardProps> = ({ id, name, primaryOnClick, secondaryOnClick }) => {
+    return (
+        <Box
+            component={"div"}
+            display={"flex"}
+            key={id}
+            justifyContent={"space-between"}
+            alignItems={"center"}
+            gap={"1rem"}
+        >
+            <SectionLabel className="text-wrap break-all">{name.replace(/.mira$/, "")}</SectionLabel>
+            <Box
+                component={"div"}
+                display={"flex"}
+                key={`button-box-${id}`}
+                flexDirection={"row-reverse"}
+                gap={"0.25rem"}
+                justifyContent={"center"}
+                alignItems={"center"}
+            >
+                <NegativeButton value={SynthesisIcons.DeleteLarge} onClick={secondaryOnClick} />
+                <PositiveButton value={SynthesisIcons.SelectLarge} onClick={primaryOnClick} />
+            </Box>
+        </Box>
+    )
+}
+
+const MatchModeConfigPanel: React.FC<PanelPropsImpl> = ({ panelId }) => {
+    const { closePanel } = usePanelControlContext()
+    const { openModal } = useModalControlContext()
+
+    const [matchModeConfigs, setMatchModeConfigs] = useState<MatchModeConfig[]>([])
+
+    useEffect(() => {
+        const loadAllJsonFiles = async () => {
+            try {
+                const indexRes = await fetch("match-mode-config/index.json")
+                const fileNames: string[] = await indexRes.json()
+
+                const defaultConfigs = await Promise.all(
+                    fileNames.map(async fileName => {
+                        const res = await fetch(`match-mode-config/${fileName}`)
+                        return res.json()
+                    })
+                )
```

Comment:
Why bother with an `index.json` file, why not just load all the cached files and all the files in `match-mode-config`?